src/ripple/app/ledger/LedgerHistory.cpp

```diff
@@ -384,7 +391,7 @@ void LedgerHistory::handleMismatch (
         log_one (validLedger, (*v)->key(), "built", j_);
 }
 
-void LedgerHistory::builtLedger (Ledger::ref ledger)
+void LedgerHistory::builtLedger (Ledger::ref ledger, Json::Value&& consensus)
 {
```

Comment:
Consider what happens when we change this to a value parameter with a typical call (line 973 of `LedgerMaster.cpp`): `mLedgerHistory.builtLedger (ledger, std::move (caller_consensus));`
`caller_consensus` will be moved, via a move constructor, to the parameter `consensus`. No deep copy is made. The move constructor is cheap for this type.

The advantage of make this a value parameter is callers do not _have_ to call it with an rvalue. I don't like an interface that forces us to move parameters. I'd like it to work for both rvalues and lvalues.

Of course, when called with an lvalue, a deep copy is made. But presumably the caller still wants to use the var, so we'd need to make a copy before we moved it with the current interface as well.